### DISCUSSION TITLE
Multiple layers of AI file are ignored by identify.exe and convert.exe

### DISCUSSION THREAD

**Discussion by yansklyarenko** [ImageMagick/ImageMagick#2247] [Help] [answered]
I have a multi-layered Adobe Illustrator file in CMYK color scheme. When I try to run `identify.exe` against it:
    `C:\ProgramData\chocolatey\bin\identify.exe example.ai`
only one layer is detected:
    `example.ai AI 42x42 42x42+0+0 16-bit ColorSeparation CMYK 7155B 0.000u 0:00.002`

The same thing happens if I try to convert it to e.g. PNG:
    `C:\ProgramData\chocolatey\bin\convert.exe example.ai example.png`
I get a PNG as while rectangle 42x42, probably, representing just one layer.

This is the sample AI file (zipped): 
[example.zip](https://github.com/ImageMagick/ImageMagick/files/4890512/example.zip)

Do you have any idea why it behaves like this? Is it expected? If so, I would appreciate any hints regarding how to transform the original file (probably with some script) so that it becomes "known" and interpreted correctly by the tools like `identify.exe` and `convert.exe`.

Sorry for the vague question - I'm still trying to narrow the problem down.
Thanks in advance!

**Comment by snibgo** [comment, marked as answer]
IM delegates rasterizing of AI files to Ghostscript. If GS can't rasterize a layer, then IM can't either.
